rio_cogeo/cogeo.py

```diff
@@ -242,6 +242,17 @@ def cog_translate(
                         rasterio.open(tmpfile.name, "w", **meta)
                     )
 
+                # Transfer color interpolation
+                ci = vrt_dst.colorinterp
+                if nodata is None and not alpha:
+                    ci = [c for c in ci if c != ColorInterp.alpha]
```

Comment:
I'm not sure to understand this part to be honest!

Reply:
I started writing an explanation, and then realized that it's not needed. I was thinking that for the cases where `add_alpha=True` when creating the VRT I needed to remove the alpha band (that doesn't seem to be the case). I'll remove it.

Reply:
so something like 
```
                if len(indexes) == 1:
                    tmp_dst.colorinterp = [ColorInterp.gray]
                else:
                    tmp_dst.colorinterp = [vrt_dst.colorinterp[b - 1] for b in indexes]

```
should be enough then